src/Tendermint/Signer.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "Serialization.h"
+
+#include "../Hash.h"
+#include "../HexCoding.h"
+#include "../Base64.h"
+#include "../PrivateKey.h"
+
+#include <google/protobuf/io/coded_stream.h>
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+#include <string>
+
+using namespace TW;
+using namespace TW::Cosmos;
+
+std::vector<uint8_t> Signer::sign() const {
+    auto key = PrivateKey(input.private_key());
+    auto hash = Hash::sha256(signaturePreimage());
+    auto signature = key.sign(hash, TWCurveSECP256k1);
+    return std::vector<uint8_t>(signature.begin(), signature.end() - 1);
+}
+
+std::string Signer::signInBase64() const {
+    auto signature = sign();
+    return Base64::encode(Data(signature.begin(), signature.end()));
+}
+
+std::string Signer::signaturePreimage() const {
+    auto json = transactionForSigningJSON(input);
+    std::cout << "Signer::signaturePreimage()" << std::endl;
+    std::cout << json.dump() << std::endl;
```

Comment:
Remove debug code

Reply:
done in ce68b6caed56b6982d81ffe3f571e9477b757ad8

Reply:
> Remove debug code

done in ce68b6caed56b6982d81ffe3f571e9477b757ad8